### PR TITLE
Unity tests of logger and devices

### PR DESCRIPTION
**Pull request by kevinuehara**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Description of PR:** 
- Creation of unity tests in python using librarys like pytest, mock, magicmock
- Refatory Handlers of DeviceManager
- Refactory connection with Kafka
- Integration of device-manager with codecov

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)

* **Other information**:

### CODE REVIEW

File: DeviceManager/DeviceHandler.py [hunk 1]
```diff
@@ -272,17 +293,17 @@ def get_devices(req, sensitive_data=False):
...
             attr_filter.append(DeviceAttr.value_type == attr_type_item)
 
         label_filter = []
-        target_label = req.args.get('label', None)
+        target_label = params.get('label')
         if target_label:
             label_filter.append(Device.label.like("%{}%".format(target_label)))
 
         template_filter = []
-        target_template = req.args.get('template', None)
```
Comment (mprevide) [inline DeviceManager/DeviceHandler.py]: Why did you remove the default?

File: DeviceManager/DeviceHandler.py [hunk 2]
```diff
@@ -328,8 +349,8 @@ def get_devices(req, sensitive_data=False):
             page = db.session.query(Device).order_by(sortBy).paginate(**pagination)
 
         devices = []
-
-        if req.args.get('idsOnly', 'false').lower() in ['true', '1', '']:
```
Comment (mprevide) [inline DeviceManager/DeviceHandler.py]: Why did you remove the default?

File: DeviceManager/ImportHandler.py [hunk 3]
```diff
@@ -26,9 +26,26 @@
 
 class ImportHandler:
 
+    kafka_handler = None
+
     def __init__(self):
         pass
 
+    @classmethod
+    def verifyInstance(cls, kafka):
```
Comment (mprevide) [inline DeviceManager/ImportHandler.py]: This method repeats maybe it could be inside BackendHandler

Comment (mprevide) [reply DeviceManager/ImportHandler.py]: And maybe it's called getInstance

Comment (mprevide) [inline DeviceManager/ImportHandler.py]: The verifyInstance method repeats in several places. Maybe it could be in one place where the handles extend a class. I do not know well.

Comment (mprevide) [reply DeviceManager/ImportHandler.py]: I don't know much about python.
But I had thought more of a parent class (for example Handler.py) where all the other somethingHandler could extend them.
And in it would remain in an attribute the instance, without the need to pass the cls, something like a singleton. But as I said, I don't know much about python and your solution seems ok to me.

**Comment by mprevide** [comment]
You could take the /test folder from the convergence statistics.

File: tests/test_backend_handler.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        KafkaHandler().update(device, meta={"service": 'admin'})
+
+    @patch("DeviceManager.BackendHandler.KafkaNotifier")
+    def test_configure_event(self, kafka_instance_mock):
+
+        device = {'templates': [369], 'label': 'test_device',
+                  'id': 1, 'created': '2019-08-29T18:18:07.801602+00:00'}
+
+        KafkaHandler().configure(device, meta={"service": 'admin'})
+
```
Comment (mprevide) [inline tests/test_backend_handler.py]: What exactly are you testing in the above 4 methods?

File: tests/test_backend_handler.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    def test_verify_intance_kafka(self):
```
Comment (mprevide) [inline tests/test_backend_handler.py]: Could you add some check to see if it really happened as expected in the above methods?

File: tests/test_import_handler.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        db_mock.session = AlchemyMagicMock()
+        self.assertIsNone(ImportHandler.restore_sequences())
+
+    def test_notifies_deletion_to_kafka(self):
+        with patch('DeviceManager.ImportHandler.serialize_full_device') as mock_serialize_device_wrapper:
+            mock_serialize_device_wrapper.return_value = {'templates': [369], 'label': 'test_device', 'id': 1,
+                                                          'created': '2019-08-29T18:18:07.801602+00:00'}
+
+            with patch.object(KafkaInstanceHandler, "getInstance", return_value=MagicMock()):
+                ImportHandler().notifies_deletion_to_kafka('test_device', 'admin')
```
Comment (mprevide) [inline tests/test_import_handler.py]: Could you see if the expected occurred?

File: tests/test_import_handler.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        self.assertIsNotNone(result)
+        self.assertTrue(result)
+
+    def test_notifies_creation_to_kafka(self):
+        with patch('DeviceManager.ImportHandler.serialize_full_device') as mock_serialize_device_wrapper:
+            mock_serialize_device_wrapper.return_value = {'templates': [369], 'label': 'test_device', 'id': 1,
+                                                          'created': '2019-08-29T18:18:07.801602+00:00'}
+
+            with patch.object(KafkaInstanceHandler, "getInstance", return_value=MagicMock()):
+                ImportHandler().notifies_creation_to_kafka(
```
Comment (mprevide) [inline tests/test_import_handler.py]: Could you see if the expected occurred?

File: tests/test_backend_handler.py (hunk 5 above)
Comment (kevinuehara) [reply tests/test_backend_handler.py]: I added a check that validates if the function passed the kafka flush in the next commit. Thanks!

File: tests/test_import_handler.py (hunk 6 above)
Comment (kevinuehara) [reply tests/test_import_handler.py]: I added a check that validates if the function passed the kafka flush remove event in the next commit. Thanks!

File: tests/test_import_handler.py (hunk 7 above)
Comment (kevinuehara) [reply tests/test_import_handler.py]: I added a check that validates if the function passed the kafka flush create event in the next commit. Thanks!

File: tests/test_backend_handler.py (hunk 5 above)
Comment (kevinuehara) [reply tests/test_backend_handler.py]: I also removed the "/test" folder from the coverage tests :)
